algorithms/src/polycommit/sonic_pc/data_structures.rs

```diff
@@ -311,12 +311,41 @@ impl<E: PairingEngine> ToBytes for CommitterKey<E> {
     }
 }
 
-impl<E: PairingEngine> CommitterKey<E> {
+/// `CommitterUnionKey` is a union of `CommitterKey`s, useful for multi-circuit batch proofs.
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+pub struct CommitterUnionKey<'a, E: PairingEngine> {
+    /// The key used to commit to polynomials.
+    pub powers_of_beta_g: Option<&'a Vec<E::G1Affine>>,
+
+    /// The key used to commit to polynomials in Lagrange basis.
+    pub lagrange_bases_at_beta_g: BTreeMap<usize, &'a Vec<E::G1Affine>>,
+
+    /// The key used to commit to hiding polynomials.
+    pub powers_of_beta_times_gamma_g: Option<&'a Vec<E::G1Affine>>,
+
+    /// The powers used to commit to shifted polynomials.
+    /// This is `None` if `self` does not support enforcing any degree bounds.
+    pub shifted_powers_of_beta_g: Option<&'a Vec<E::G1Affine>>,
+
+    /// The powers used to commit to shifted hiding polynomials.
+    /// This is `None` if `self` does not support enforcing any degree bounds.
+    pub shifted_powers_of_beta_times_gamma_g: Option<BTreeMap<usize, &'a Vec<E::G1Affine>>>,
+
+    /// The degree bounds that are supported by `self`.
+    /// Sorted in ascending order from smallest bound to largest bound.
+    /// This is `None` if `self` does not support enforcing any degree bounds.
+    pub enforced_degree_bounds: Option<Vec<usize>>,
```

Comment:
Why does these need to be wrapped in an `Option`? Also, we can resolve the difference between `CommitterKey` and `CommitterUnionKey` by using `Cow`, which inherently abstracts over values and references.

Reply:
> Why does these need to be wrapped in an Option?

This was already set up this way, and at least some tests exist which set this to `None`, maybe just to test edge cases for the polynomial commitment scheme.

> Also, we can resolve the difference between CommitterKey and CommitterUnionKey by using Cow, which inherently abstracts over values and references.

Ah that is great thx for *pointing* out (pun intended)